brainclouds2s.js

```diff
@@ -29,60 +27,47 @@ function s2sRequest(context, json, callback)
         }
     }
 
-    var req = https.request(options, res =>
-    {
+    var req = https.request(options, res => {
         var data = ''
-        
+
         // A chunk of data has been recieved.
-        res.on('data', chunk =>
-        {
+        res.on('data', chunk => {
             data += chunk
         })
-        
+
         // The whole response has been received. Print out the result.
-        res.on('end', () =>
-        {
-            if (context.logEnabled)
-            {
+        res.on('end', () => {
+            if (context.logEnabled) {
                 console.log(`[S2S RECV ${context.appId}] ${data}`)
             }
-            if (callback)
-            {
-                if (data)
-                {
-                    
+            if (callback) {
+                if (data) {
+
                     try {
                         let dataJson = JSON.parse(data)
-                        callback(context, dataJson)
-                    } 
-                    catch (error) 
-                    {
-                        if (context.logEnabled)
-                        {
+                        callCallback(callback, context, dataJson)
```

Comment:
Hmm. I think you should move this call outside the try/catch... i.e. only use this try/catch to guard against errors when trying to parse the response data. That way, you can make the error message more explicit (i.e. "Failed to parse response" or something like that) and you can call callCallback without it being inside a try/catch (since it does its own try/catch internally).